### PR TITLE
Add support for Scala 2.11

### LINKED ISSUE #1: Scala 2.11 support?
should we support Scala 2.11 here? (would it be hard?)

in the library ecosystem, I think a lot of libraries will be cross-building to 2.11 for a long while yet

### PR DESCRIPTION
**Pull request by julienrf**
Fixes #1 

The only required change was that in `ImmutableArray`’s implementation I had to inline a couple of methods that are defined in 2.12 but don’t exist in 2.11.

### CODE REVIEW

**Comment by SethTisue** [comment]
do we really want to support 2.10? I think most libraries are dropping 2.10 support in their most recent versions, so I think the cost/benefit here might be poor. (and perhaps the cost looks low now but will turn out to be higher later...)

**Comment by SethTisue** [comment]
not sure we should support 2.10. let's discuss further on the issue

File: build.sbt [hunk 1]
```diff
@@ -6,10 +6,12 @@ version := "0.1-SNAPSHOT"
...
 
-unmanagedSourceDirectories in Compile ++=
-  (if(scalaVersion.value.startsWith("2.13.")) Seq((sourceDirectory in Compile).value / "scala-2.13") else Seq())
+unmanagedSourceDirectories in Compile += (
+  if(scalaVersion.value.startsWith("2.13.")) (sourceDirectory in Compile).value / "scala-2.13"
+  else (sourceDirectory in Compile).value / "scala-2.10_2.12"
+)
 
-crossScalaVersions := Seq("2.12.4", "2.13.0-pre-c577876")
+crossScalaVersions := Seq("2.10.7", "2.12.5", "2.11.12", "2.13.0-pre-c577876")
```
[the remark this answers is not in the document]
Comment (SethTisue) [reply build.sbt]: I don't see any value in providing 2.10 support that might be dropped any time. I see negative value in this, in fact, as it leads potential users down a garden path of expecting continued 2.10 support.

[the remark this answers is not in the document]
Comment (julienrf) [reply build.sbt]: scalacheck is heavily depended on and it currently supports Scala 2.10. If we drop 2.10 support in the compat library, either scalacheck and all the projects that depend on will have to drop 2.10 support, or we will have to copy the contents of the compat library in scalacheck.

**Comment by julienrf** [comment]
I’ve removed 2.10 support because some tests were failing due to a difference in type inference.

**Pull request merged**